apps/ledger-live-mobile/src/newArch/features/Web3Hub/screens/Web3HubTabs/components/TabItem/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { Image, StyleSheet, TouchableOpacity, View } from "react-native";
+import { Flex, IconsLegacy, Text } from "@ledgerhq/native-ui";
+import { useTheme } from "@react-navigation/native";
+import type { TabData, TabsProps } from "LLM/features/Web3Hub/types";
+import React, { useCallback, useState } from "react";
+import { NavigatorName, ScreenName } from "~/const";
+
+export default function TabItem({
+  item,
+  navigation,
+  onItemClosePress,
+}: {
+  item: TabData;
+  navigation: TabsProps["navigation"];
+  onItemClosePress: (itemId: string) => void;
+}) {
+  const [imageLoaded, setImageLoaded] = useState(true);
+  const handleImageLoad = useCallback(() => setImageLoaded(true), []);
+  const handleImageError = useCallback(() => setImageLoaded(false), []);
+  const { colors } = useTheme();
+
+  const goToApp = useCallback(() => {
+    navigation.push(NavigatorName.Web3Hub, {
+      screen: ScreenName.Web3HubApp,
+      params: {
+        manifestId: item?.manifestId,
+      },
+    });
+  }, [navigation, item?.manifestId]);
+
+  const handleClosePress = useCallback(() => {
+    onItemClosePress(item.id);
+  }, [item.id, onItemClosePress]);
+
+  return (
+    <TouchableOpacity
+      testID={`web3hub-tab-item-${item.manifestId}`}
+      style={{ flex: 1 }}
+      onPress={goToApp}
+    >
+      <View style={styles.container}>
+        <Flex
+          flexDirection="row"
+          backgroundColor={colors.background}
+          height={30}
+          alignItems={"center"}
+          paddingX={3}
+          columnGap={5}
+        >
+          {imageLoaded && item.icon ? (
+            <Image
+              source={{
+                uri: item.icon,
+              }}
+              style={styles.icon}
+              onLoad={handleImageLoad}
+              onError={handleImageError}
+            />
+          ) : (
+            <View style={{ ...styles.icon, backgroundColor: colors.black }} />
+          )}
+
+          <Text style={{ ...styles.title, flex: 1, color: "#C3C3C3" }}>{item.title}</Text>
+
+          <TouchableOpacity onPress={handleClosePress}>
+            <IconsLegacy.CloseMedium size={16} color={"#C3C3C3"} />
+          </TouchableOpacity>
+        </Flex>
+
+        <View style={{ flex: 1, height: 30, backgroundColor: "white" }}>
+          <Image source={{ uri: item.previewUri }} style={{ width: "100%", height: "100%" }} />
+        </View>
+      </View>
+    </TouchableOpacity>
+  );
+}
+
+const styles = StyleSheet.create({
```

Comment:
it's not a recommended/readable way to use of variables before they are defined, can you move this styles on top of the component's script please ?

Reply:
On mobile with RN we usually place the styles at the bottom to have the component first, not a big deal in JS and quite helpful for readability IMO

Reply:
it's not big deal while all variable are hoisted i agree with you , but it's a little bit confusing (less readable and anti `no-use-before-define`rule. (it's a very optional recommendation)